src/common/proc.c

```diff
@@ -47,6 +50,21 @@ crt_proc_prop_entries(crt_proc_t proc, daos_prop_t *prop)
 		if (entry->dpe_type == DAOS_PROP_PO_LABEL ||
 		    entry->dpe_type == DAOS_PROP_CO_LABEL)
 			rc = crt_proc_d_string_t(proc, &entry->dpe_str);
+		else if (entry->dpe_type == DAOS_PROP_PO_ACL ||
+			 entry->dpe_type == DAOS_PROP_CO_ACL) {
+			if (entry->dpe_val_ptr == NULL) {
+				memset(&iov, 0, sizeof(iov));
+			} else {
+				acl = (struct daos_acl *)entry->dpe_val_ptr;
+				d_iov_set(&iov, entry->dpe_val_ptr,
+					daos_acl_get_size(acl));
+			}
+
+			rc = crt_proc_d_iov_t(proc, &iov);
+			if (rc == 0 && entry->dpe_val_ptr == NULL) {
```

Comment:
better to check proc is for decode and assign the dpe_val_ptr, and set to NULL when PROC_FREE.
crt_proc_op_t	proc_op;
rc = crt_proc_get_op(proc, &proc_op);
if (rc)
        break;
......

if (rc)
break;
if (proc_op == CRT_PROC_DECODE)
        entry->dpe_val_ptr = iov.iov_buf;
else if (proc_op == CRT_PROC_FREE)
        entry->dpe_val_ptr = NULL;

Reply:
BTW, defining another "crt_proc_struct_daos_acl" probably is clearer, although using the crt_proc_d_iov_t should work also.

Reply:
Fixed and moved this processing logic into its own helper function